### ISSUE TITLE
Plugin V2: Move scheduling information from notes to dedicated JSON file

### ISSUE THREAD

**Issue by st3v3nmw** [st3v3nmw/obsidian-spaced-repetition#153]
TL;DR

I wanna move the scheduling information from inside the notes to a dedicated JSON file. Reasons include speedup, better statistics, & support for more features.

## Version 2

Just store a unique identifier & the scheduling information is stored in the JSON file.

`<!--SRID: UUID code-->` i.e. `<!--SRID: nCc1xigS0lh5qNTKapAxN-->`

The JSON file will only contain the scheduling information (i.e. it won't store any card text) hence making the .md notes remain as the authoritative source for the card text.

### Advantages

- Speed i.e. currently, the plugin has to scan _**all**_ tagged notes to check the due dates. This is particularly slow on the phone (I do my cards there).
- Tracking of deck-level statistics like retention, card maturity, and answer button clicks which allows the user to tweak the algorithm settings to suit their needs.
- Much cleaner syntax.
- Better replication of the Anki algorithm i.e. learning stages, card suspension, tracking leeches.
- Certain open issues can be solved better this way i.e. #86 (#93, #34, #43, #49, #58, #80), #109
- Custom flashcards styles.
- Can make the algorithm user configurable too (long term).

### Legacy support

- Command to move everything to version 2 without losing anything
- Expose all scheduling information to the user (by displaying it inside the notes next to the card)
